benchmark/scripts/consolidate.ts

```diff
@@ -0,0 +1,337 @@
+
+// Utils to extract values from log messages
+const isoDateRegex = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/;
+const requestIdRegex = /RequestId: (\S+)/;
+const durationRegex = /Duration: ([0-9.]+) ms/;
+const maxMemoryRegex = /Memory Used: ([0-9]+) MB/;
+const initDurationRegex = /Init Duration: ([0-9.]+) ms/;
+function extract(regex: RegExp, message: string): string | undefined {
+  const match = regex.exec(message);
+  return match && match[1] ? match[1] : undefined;
+}
+const parseNumber = (value?: string) => (value ? parseFloat(value) : undefined);
+const getJson = (value: string) => {
+  const jsonStartIndex = value.indexOf("{");
+  const jsonEndIndex = value.lastIndexOf("}") + 1;
+  const jsonStr = value.slice(jsonStartIndex, jsonEndIndex);
+  const json = JSON.parse(jsonStr);
+  return json;
+};
+
+// Accessing nested objects by string path
+// https://stackoverflow.com/questions/6491463/accessing-nested-javascript-objects-and-arrays-by-string-path
+const setPath = (object: any, path: string, value: any) =>
+  path
+    .split(".")
+    .reduce(
+      (o, p, i) => (o[p] = path.split(".").length === ++i ? value : o[p] || {}),
+      object
+    );
+const resolvePath = (object: any, path: string, defaultValue: any) =>
+  path.split(".").reduce((o, p) => (o ? o[p] : defaultValue), object);
+
+async function main() {
+  const start = performance.now();
+  // Setup
+  const gitBranch = execSync("git rev-parse --abbrev-ref HEAD 2>/dev/null", {
+    encoding: "utf-8",
+  }).trim();
+  let gitTag: string | undefined;
+  const rawFilePath = path.join(
+    __dirname,
+    CONFIG.logs.outputFolder,
+    `${gitBranch}/raw.json`
+  );
+  const filePath = path.join(
+    __dirname,
+    CONFIG.logs.outputFolder,
+    `${gitBranch}/consolidated.json`
+  );
+
+  console.log(
+    `Consolidate logs for branch '${gitBranch}' ${gitTag ? `(${gitTag})` : ""}`
+  );
+
+  console.log("\n- Delete previous artifacts if they exist");
+  // Delete if exist
+  // - previous results files
+  try {
+    // console.log(`  - Previous result file: '${filePath}'`);
+    await fsPromises.unlink(filePath);
+  } catch {
+    // console.log(`    - not found`);
+  }
+
+  console.log(`- Read raw data`);
+  const raw: IResult = JSON.parse(
+    await fsPromises.readFile(rawFilePath, "utf8")
+  );
+  let result: IResult = {
+    git: { ...raw.git },
+    time: { ...raw.time },
+    datasets: {
+      coldStarts: {
+        initDuration: {},
+        duration: {},
+        totalDuration: {},
+        maxMemory: {},
+        httpRequestLatency: {},
+        apiCallLatency: {},
+      },
+      warmStarts: {
+        duration: {},
+        maxMemory: {},
+        httpRequestLatency: {},
+        apiCallLatency: {},
+      },
+    },
+    log: [],
+  };
+
+  console.log(`- Parse the log`);
+  let count = { functionCalls: 0, logEntries: 0 };
+  let consolidatedEntry: Record<string, any> = {};
+  let requestId: string | undefined;
+  for (const entry of raw.log) {
```

Comment:
some comments might be helpful - it looks like this is aggregating the results of multiple runs into a single result?